x-pack/plugins/infra/public/alerting/metric_threshold/components/custom_equation/custom_equation_editor.tsx

```diff
@@ -0,0 +1,211 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import {
+  EuiFieldText,
+  EuiFormRow,
+  EuiFlexItem,
+  EuiFlexGroup,
+  EuiButtonEmpty,
+  EuiSpacer,
+} from '@elastic/eui';
+import React, { useState, useCallback, useMemo } from 'react';
+import { omit, range, first, xor, debounce } from 'lodash';
+import { IErrorObject } from '@kbn/triggers-actions-ui-plugin/public';
+import { FormattedMessage } from '@kbn/i18n-react';
+import { OMITTED_AGGREGATIONS_FOR_CUSTOM_METRICS } from '../../../../../common/http_api';
+import {
+  Aggregators,
+  CustomMetricAggTypes,
+  MetricExpressionCustomMetric,
+} from '../../../../../common/alerting/metrics';
+import { MetricExpression } from '../../types';
+import { CustomMetrics, AggregationTypes, NormalizedFields } from './types';
+import { MetricRowWithAgg } from './metric_row_with_agg';
+import { MetricRowWithCount } from './metric_row_with_count';
+import {
+  CUSTOM_EQUATION,
+  EQUATION_HELP_MESSAGE,
+  LABEL_HELP_MESSAGE,
+  LABEL_LABEL,
+} from '../../i18n_strings';
+
+export interface CustomEquationEditorProps {
+  onChange: (expression: MetricExpression) => void;
+  expression: MetricExpression;
+  fields: NormalizedFields;
+  aggregationTypes: AggregationTypes;
+  errors: IErrorObject;
+}
+
+const NEW_METRIC = { name: 'A', aggType: Aggregators.AVERAGE as CustomMetricAggTypes };
+const VAR_NAMES = range(65, 65 + 26).map((c) => String.fromCharCode(c));
+
+export const CustomEquationEditor: React.FC<CustomEquationEditorProps> = ({
+  onChange,
+  expression,
+  fields,
+  aggregationTypes,
+  errors,
+}) => {
```

Comment:
nit: Since we're not using `children`, I guess we could remove the usage of `React.FC` and just go for a functional component. WDYT?

```ts
export const CustomEquationEditor = ({
  onChange,
  expression,
  fields,
  aggregationTypes,
  errors,
}: CustomEquationEditorProps) => {
```